WordPress/Classes/ViewRelated/Gutenberg/GutenbergMediaInserterHelper.swift

```diff
@@ -153,7 +153,21 @@ class GutenbergMediaInserterHelper: NSObject {
             guard let urlString = media.remoteURL, let url = URL(string: urlString), let mediaServerID = media.mediaID?.int32Value else {
                 break
             }
-            gutenberg.mediaUploadUpdate(id: mediaUploadID, state: .succeeded, progress: 1, url: url, serverID: mediaServerID)
+            if media.mediaType == .image {
+                gutenberg.mediaUploadUpdate(id: mediaUploadID, state: .succeeded, progress: 1, url: url, serverID: mediaServerID)
+            } else if media.mediaType == .video {
+                fetchVideoPressUrl(for: media) { [weak self] (result) in
```

Comment:
VideoPress information will only be available if the site is WordPress.com or using Jetpack. 
If it's a self-hosted site without Jetpack the video will not be videoPress and you will need to send the 'media.remoteURL' directly.

Reply:
great catch! I have updated the code accordingly